### PR TITLE
Feature/gdpr fixes

### PR DESCRIPTION
**Pull request by gymad**
This PR contains some fixes for related GDRP PR #4911 

## Description

- fixing and remove php notices from apache log, for e.g at login screen and using default theme
- Add APP_LIST_STRINGS, APP, MOD automatically for each Sugar_Smarty templates
- code refactoring on opt in email sending

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: include/SugarObjects/templates/basic/include/ConfirmOptIn.php [hunk 1]
```diff
@@ -73,11 +73,8 @@ public function pre_display()
      */
     public function display()
     {
-        global $app_list_strings, $app_strings, $mod_strings;
+
```
Comment (daniel-samson) [inline include/SugarObjects/templates/basic/include/ConfirmOptIn.php]: We need these values, so that third party developers can customise the template.

Comment (gymad) [reply include/SugarObjects/templates/basic/include/ConfirmOptIn.php]: they can

Comment (zoltankocsardi) [reply include/SugarObjects/templates/basic/include/ConfirmOptIn.php]: These values are not used. If someone wants to use it, he'll add that line above.

**Comment by Dillon-Brown** [comment]
Assessed

**Pull request merged**